sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -12,18 +15,60 @@ namespace core
 namespace http
 {
 
-class http_request
+namespace http_method
+{
+enum HttpMethod
+{
+  GET,
+  HEAD,
+  POST,
+  PUT,
+  DELETE,
+  PATCH,
+};
+}
+
+class Request
 {
 
 private:
-  /* data */
+  Request(Request const&) = delete;
+  void operator=(Request const&) = delete;
+
+  http_method::HttpMethod _method;
+  std::string _url;
+  std::map<std::string, std::string> _headers;
+  std::string _body;
+  size_t _query_start; // 0 = no query in url, > 0 = query start position in url '?'
+
+  static size_t get_query_start(std::string const& url)
+  {
+    auto position = url.find('?');
+    return position == std::string::npos ? 0 : position;
+  }
+
 public:
-  http_request();
-  ~http_request();
-};
+  Request(http_method::HttpMethod httpMethod, std::string const& url)
+      : _method(std::move(httpMethod)), _url(std::move(url)),
+        _query_start(get_query_start(std::move(url)))
+  {
+  }
 
-http_request::http_request() {}
-http_request::~http_request() {}
+  Request(http_method::HttpMethod httpMethod, std::string const& url, std::string const& body)
+      : Request(httpMethod, url)
+  {
+    this->_body = body;
+  }
+
+  http_method::HttpMethod getMethod();
+  std::string const& getUrl();
```

Comment:
BTW, all these methods that don't modify the state of the object (normally, getters), we should declare as `std::string const& getUrl() const`. So that when you have a const pointer/reference to http_request, you still can invoke getUrl(). Sorry for missing it earlier.